libtenzir/builtins/operators/to_opensearch.cpp

```diff
@@ -0,0 +1,538 @@
+      auto id = ids ? values(ty, as<arrow::StringArray>(*ids->array)) : ng();
+      auto idx = idxs ? values(ty, as<arrow::StringArray>(*idxs->array)) : ng();
+      auto act = acts ? values(ty, as<arrow::StringArray>(*acts->array)) : ng();
+      for (auto&& doc : docs.values()) {
+        const auto ptr = try_as<view<record>>(doc);
+        const auto action = act.next();
+        const auto actual_id = id.next();
+        const auto actual_idx = idx.next();
+        if (not ptr) {
+          diagnostic::warning("`doc` evaluated to non-record, skipping event")
+            .primary(*args_.doc)
+            .emit(dh);
+          continue;
+        }
+        if (not action or not action.value()) {
+          diagnostic::warning("`action` evaluated to `null`, skipping event")
+            .primary(args_.action)
+            .emit(dh);
+          continue;
+        }
+        if (auto diag
+            = b.create_metadata(**action, actual_idx, actual_id, args_)) {
+          dh.emit(std::move(diag).value());
+          continue;
+        }
+        b.create_doc(**action, *ptr);
+        switch (b.finish_event()) {
+          using enum json_builder::state;
+          case ok: {
+            break;
+          }
+          case full: {
+            send_req(req, b.yield(dh), dh);
+            break;
+          }
+          case event_too_large: {
+            diagnostic::warning(
+              "event too large for given `max_content_length`")
+              .note("serialized event size was `{}`", b.last_element_size())
+              .primary(*args_.max_content_length)
+              .emit(dh);
+            break;
+          }
+        }
+      }
+    }
+    if (b.has_contents()) {
+      send_req(req, b.yield(dh), dh);
+    }
+  }
+
+  auto
+  optimize(expression const&, event_order) const -> optimize_result override {
+    return do_not_optimize(*this);
+  }
+
+  auto name() const -> std::string override {
+    return "to_opensearch";
+  }
+
+  auto location() const -> operator_location override {
+    return operator_location::local;
+  }
+
+  friend auto inspect(auto& f, opensearch_operator& x) -> bool {
+    return f.apply(x.args_);
+  }
+
+  auto detached() const -> bool override {
+    return true;
+  }
+
+private:
+  opensearch_args args_;
+};
+
+struct plugin : public virtual operator_plugin2<opensearch_operator> {
+  auto name() const -> std::string override {
+    return "to_opensearch";
+  }
+
+  auto
+  make(invocation inv, session ctx) const -> failure_or<operator_ptr> override {
+    auto args = opensearch_args{};
+    auto p = argument_parser2::operator_(name());
+    args.add_to(p);
+    TRY(p.parse(inv, ctx));
+    if (not args.validate(ctx)) {
+      return failure::promise();
+    }
+    return std::make_unique<opensearch_operator>(std::move(args));
+  }
+
+  auto save_properties() const -> save_properties_t override {
+    return {.schemes = {"elasticsearch", "opensearch"}, .strip_scheme = true};
```

Comment:
My bad, but this is actually missing `.events=true`.

Reply:
Added, please review.